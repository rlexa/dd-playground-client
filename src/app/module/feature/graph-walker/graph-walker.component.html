<ng-template #tmplNode let-item let-header="header">
  <div class="contenthv childhv hovered-elevated list-v">
    <h4 *ngIf="header" class="text-center">{{ header }}</h4>
    <div class="bar-h">
      <label class="childh">{{ item.inc.length }}</label>
      <label class="childh text-center" style="flex: 1 0 auto">{{ 'inc .. out' }}</label>
      <label class="childh">{{ item.out.length }}</label>
    </div>
    <app-simple-view [data]="item.data" [isDense]="true" [isWidthConstrained]="false" />
  </div>
</ng-template>

<ng-template #tmplLink let-item let-header="header">
  <div class="contenthv childhv hovered-elevated list-v">
    <h4 *ngIf="header" class="text-center">{{ header }}</h4>
    <app-simple-view [data]="item.data" [isDense]="true" [isWidthConstrained]="false" />
  </div>
</ng-template>

<!-- START HERE -->

<mat-card appearance="outlined">
  <mat-card-content>
    <div class="list-v">
      <div class="list-h wrap">
        <div class="contenthv childhv hovered-elevated">
          <h2>DB Stats: Overall</h2>
          <app-simple-view [data]="dbState$ | async" [isDense]="true" [isWidthConstrained]="false" />
        </div>

        <div class="contenthv childhv hovered-elevated">
          <h2>DB Stats: Node Types</h2>
          <app-simple-view [data]="dbNodeTypeCount$ | async" [isDense]="true" [isWidthConstrained]="false" />
        </div>
      </div>

      <div class="contenthv childhv hovered-elevated bar-h wrap">
        <label>Query first node of:</label>

        <mat-form-field class="margin-left childhv" hintLabel="Type">
          <mat-select [value]="curType$ | async" (selectionChange)="setCurType($event.value)">
            <mat-option *ngFor="let ii of dbTypes$ | async" [value]="ii">{{ ii }}</mat-option>
          </mat-select>
        </mat-form-field>

        <mat-form-field class="margin-left childhv" hintLabel="Tag">
          <mat-select [value]="curTag$ | async" (selectionChange)="setCurTag($event.value)">
            <mat-option *ngFor="let ii of curTypeKeys$ | async" [value]="ii">{{ ii }}</mat-option>
          </mat-select>
        </mat-form-field>

        <button *ngFor="let ii of CMPS" mat-mini-fab (click)="tryQuery(ii)">{{ ii }}</button>

        <mat-form-field class="margin-left childhv" hintLabel="Value">
          <input
            #valInput
            matInput
            [value]="curVal$ | async"
            (change)="setCurVal(valInput.value)"
            [matAutocomplete]="valAutoComplete"
            (focus)="setCurVal(valInput.value)"
            (keyup)="setCurVal(valInput.value)"
            (keyup.enter)="tryQuery()"
          />
          <mat-autocomplete #valAutoComplete="matAutocomplete" (optionSelected)="setCurVal($event.option.value)">
            <mat-option *ngFor="let ii of curTypeKeyTagValuesFiltered$ | async" [value]="ii">{{ ii }}</mat-option>
          </mat-autocomplete>
        </mat-form-field>
      </div>

      <div *ngIf="curNode$ | async as node" [ngStyle]="{display: 'grid', 'grid-template-areas': GRID, 'grid-template-columns': GRID_SIZE}">
        <div [ngStyle]="{'grid-area': GRID_FROM}">
          <div class="list-v" [style.align-items]="'flex-end'">
            <div *ngFor="let link of node.inc" class="childh list-h">
              <ng-container *ngTemplateOutlet="tmplNode; context: {$implicit: link.from}" />
              <div class="bar-h">
                <ng-container *ngTemplateOutlet="tmplLink; context: {$implicit: link}" />
                <button mat-mini-fab (click)="setCurNode(link.from)">
                  <mat-icon>{{ 'go_left' | icon }}</mat-icon>
                </button>
              </div>
            </div>
          </div>
        </div>

        <div [ngStyle]="{'grid-area': GRID_CURRENT}">
          <div class="childh bar-v">
            <ng-container *ngTemplateOutlet="tmplNode; context: {$implicit: node, header: 'CURRENT'}" />
          </div>
        </div>

        <div [ngStyle]="{'grid-area': GRID_TO}">
          <div class="childh list-v">
            <div *ngFor="let link of node.out" class="list-h">
              <div class="bar-h">
                <button mat-mini-fab (click)="setCurNode(link.to)">
                  <mat-icon>{{ 'go_right' | icon }}</mat-icon>
                </button>
                <ng-container *ngTemplateOutlet="tmplLink; context: {$implicit: link}" />
              </div>
              <ng-container *ngTemplateOutlet="tmplNode; context: {$implicit: link.to}" />
            </div>
          </div>
        </div>
      </div>
    </div>
  </mat-card-content>
</mat-card>
