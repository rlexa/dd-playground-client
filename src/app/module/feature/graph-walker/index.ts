export * from './graph-walker.component';
