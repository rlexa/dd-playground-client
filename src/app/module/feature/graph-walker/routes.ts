import {Route} from '@angular/router';
import {RouteRoot, RouteWild} from 'src/app/routing';

export default [
  {path: RouteRoot, loadComponent: () => import('./graph-walker.component').then((ii) => ii.GraphWalkerComponent)},
  {path: RouteWild, redirectTo: RouteRoot},
] as Route[];
