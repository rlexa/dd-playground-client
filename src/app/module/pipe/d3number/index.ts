export * from './d3number.pipe';
