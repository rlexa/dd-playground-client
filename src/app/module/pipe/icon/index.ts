export * from './icon-pipe.module';
