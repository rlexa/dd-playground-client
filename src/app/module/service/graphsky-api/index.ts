export { IGraphskyData, IGraphskyLinkRequest, IGraphskyNode } from './graphsky';
export * from './graphsky-api.module';
export * from './graphsky.service';
