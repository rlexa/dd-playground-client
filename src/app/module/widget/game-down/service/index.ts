export * from './game-down.service';
