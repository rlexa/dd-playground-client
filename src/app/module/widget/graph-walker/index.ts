export * from './graph-walker.component';
export * from './graph-walker.module';
