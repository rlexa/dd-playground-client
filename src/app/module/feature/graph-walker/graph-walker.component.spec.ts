import {ComponentFixture} from '@angular/core/testing';
import {MockBuilder, MockRender} from 'ng-mocks';
import {GraphskyService} from 'src/app/module/service/graphsky-api';
import {IconPipe} from '../../pipe/icon';
import {GraphWalkerComponent} from './graph-walker.component';

describe('GraphWalkerComponent', () => {
  let fixture: ComponentFixture<GraphWalkerComponent>;

  beforeEach(() =>
    MockBuilder(GraphWalkerComponent)
      .mock(IconPipe, (val) => `icon ${val}`)
      .provide(GraphskyService),
  );

  beforeEach(() => {
    fixture = MockRender(GraphWalkerComponent);
    fixture.detectChanges();
  });

  test('renders', () => expect(fixture).toMatchSnapshot());
});
