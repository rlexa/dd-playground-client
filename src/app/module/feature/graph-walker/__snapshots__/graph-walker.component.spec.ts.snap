// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphWalkerComponent renders 1`] = `
<mock-render>
  <app-graph-walker>
    <mat-card
      appearance="outlined"
      ng-reflect-appearance="outlined"
    >
      <mat-card-content>
        <div
          ddflexbox="vyt"
          ng-reflect-dd-flexbox="vyt"
        >
          <div
            ddflexbox="hysw"
            ng-reflect-dd-flexbox="hysw"
          >
            <div
              class="contenthv childhv hovered-elevated"
            >
              <h2>
                DB Stats: Overall
              </h2>
              <app-simple-view
                ng-reflect-data="[object Object]"
                ng-reflect-is-dense="true"
                ng-reflect-is-width-constrained="false"
              />
            </div>
            <div
              class="contenthv childhv hovered-elevated"
            >
              <h2>
                DB Stats: Node Types
              </h2>
              <app-simple-view
                ng-reflect-is-dense="true"
                ng-reflect-is-width-constrained="false"
              />
            </div>
          </div>
          <div
            class="contenthv childhv hovered-elevated"
            ddflexbox="hycw"
            ng-reflect-dd-flexbox="hycw"
          >
            <label>
              Query first node of:
            </label>
            <mat-form-field
              class="margin-left childhv"
              hintlabel="Type"
              ng-reflect-hint-label="Type"
            >
              <mat-select>
              </mat-select>
            </mat-form-field>
            <mat-form-field
              class="margin-left childhv"
              hintlabel="Tag"
              ng-reflect-hint-label="Tag"
            >
              <mat-select>
              </mat-select>
            </mat-form-field>
            <button
              mat-mini-fab=""
            >
              &lt;
            </button>
            <button
              mat-mini-fab=""
            >
              &lt;=
            </button>
            <button
              mat-mini-fab=""
            >
              !=
            </button>
            <button
              mat-mini-fab=""
            >
              =
            </button>
            <button
              mat-mini-fab=""
            >
              &gt;=
            </button>
            <button
              mat-mini-fab=""
            >
              &gt;
            </button>
            <mat-form-field
              class="margin-left childhv"
              hintlabel="Value"
              ng-reflect-hint-label="Value"
            >
              <input
                matinput=""
                ng-reflect-autocomplete="[object Object]"
              />
              <mat-autocomplete>
              </mat-autocomplete>
            </mat-form-field>
          </div>
        </div>
      </mat-card-content>
    </mat-card>
  </app-graph-walker>
</mock-render>
`;
